cradle-core/src/main/java/com/exactpro/cradle/serialization/MessageSerializer.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright 2021-2022 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.cradle.serialization;
+
+import com.exactpro.cradle.messages.StoredMessage;
+import com.exactpro.cradle.messages.StoredMessageId;
+import com.exactpro.cradle.messages.StoredMessageMetadata;
+
+import java.nio.ByteBuffer;
+import java.util.Collection;
+import java.util.Map;
+
+import static com.exactpro.cradle.serialization.Serialization.MessageBatchConst.*;
+import static com.exactpro.cradle.serialization.SerializationUtils.printBody;
+import static com.exactpro.cradle.serialization.SerializationUtils.printInstant;
+import static com.exactpro.cradle.serialization.SerializationUtils.printString;
+
+public class MessageSerializer {
+
+	public byte[] serializeBatch(Collection<StoredMessage> batch) throws SerializationException {
+		SerializationBatchSizes messageBatchSizes = MessagesSizeCalculator.calculateMessageBatchSize(batch);
+		ByteBuffer buffer = ByteBuffer.allocate(messageBatchSizes.total);
+		
+		this.serializeBatch(batch, buffer, messageBatchSizes);
+		
+		return buffer.array();
+	}
+
+	public void serializeBatch(Collection<StoredMessage> batch, ByteBuffer buffer, SerializationBatchSizes messageBatchSizes) throws SerializationException {
+
+		if (messageBatchSizes == null) {
+			messageBatchSizes = MessagesSizeCalculator.calculateMessageBatchSize(batch);
+		}
+		
+		buffer.putInt(MESSAGE_BATCH_MAGIC);
+		buffer.put(MESSAGE_PROTOCOL_VER);
+		
+		buffer.putInt(batch.size());
+		int i = 0;
+		for (StoredMessage message : batch) {
+			buffer.putInt(messageBatchSizes.entities[i]);
+			this.serialize(message, buffer);
+			i++;
+		}
+	}
+	
+	public byte[] serialize(StoredMessage message) throws SerializationException {
+		ByteBuffer b = ByteBuffer.allocate(MessagesSizeCalculator.calculateMessageSize(message));
+		this.serialize(message, b);
+		return b.array();
+	}
+
+	public void serialize(StoredMessage message, ByteBuffer buffer) throws SerializationException {
+		buffer.putShort(MESSAGE_MAGIC);
+		StoredMessageId id = message.getId();
+		buffer.putLong(id.getSequence());
```

Comment:
Usually we show timestamp before sequence, but this is not important in serialization/ Just wanted to mention.